test/unit/argument_parser/format_parse_test.cpp

```diff
@@ -936,3 +926,37 @@ TEST(parse_test, subcommand_argument_parser_success)
                      seqan3::argument_parser_error);
     }
 }
+
+TEST(parse_test, issue1544)
+{
+    {   // unknown option that has a prefix of a known option
+        std::string option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(option_value, 'f', "foo", "this is a string option.");
+
+        EXPECT_THROW(parser.parse(), seqan3::unknown_option);
+    }
+
+    {   // known option that has a prefix of a unknown option
+        std::string option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(option_value, 'f', "foo-bar", "this is a string option.");
+
+        EXPECT_THROW(parser.parse(), seqan3::unknown_option);
+    }
+
+    {   // known option is a prefix of another known option
+        std::string foo_option_value;
+        std::string foobar_option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(foo_option_value, 'f', "foo", "this is a prefix of foobar.");
+        parser.add_option(foobar_option_value, 'b', "foo-bar", "this has prefix foo.");
+
+        EXPECT_NO_THROW(parser.parse());
+        EXPECT_EQ(foo_option_value, "hallo");
+        EXPECT_EQ(foobar_option_value, "ballo");
+    }
```

Comment:
Does this also work when mixing short and long identifier. Basically, when I call  `-f` and `--foo-bar`?

Reply:
No because I explicitly test for the single `-` or double dash `--` so those are never amigious.

Reply:
amigious 🤠